e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -113,16 +144,18 @@ def run_gc_checkpoints_test(checkpoint_storage: Dict[str, str]) -> None:
                 storage_manager = storage.build(checkpoint_config, container_path=None)
                 storage_state = {}  # type: Dict[str, Any]
                 for checkpoint in checkpoints:
-                    storage_id = checkpoint["uuid"]
+                    if not checkpoint.uuid:
```

Comment:
for some reason it believes the type could be None (`error: Invalid index type "Optional[str]" for "Dict[str, Any]"; expected type "str"`), so I want to filter out this possibility.  I think this could be due to no typing on `all_checkpoints = []`

Reply:
I'm going to add typing earlier, and replace this if/continue with an `assert checkpoint.uuid`